Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h

```diff
@@ -1801,70 +1921,63 @@ std::size_t duplicate_non_manifold_vertices(TriangleMesh& tm,
     = choose_param(get_param(np, internal_np::output_iterator),
                    Emptyset_iterator());
 
-  internal::Vertex_collector<TriangleMesh, Output_iterator> dmap;
+  internal::Vertex_collector<PolygonMesh> dmap;
   boost::unordered_set<vertex_descriptor> vertices_handled;
   boost::unordered_set<halfedge_descriptor> halfedges_handled;
 
   std::size_t nb_new_vertices = 0;
 
   std::vector<halfedge_descriptor> non_manifold_cones;
-  BOOST_FOREACH(halfedge_descriptor h, halfedges(tm))
+  BOOST_FOREACH(halfedge_descriptor h, halfedges(pm))
   {
     // If 'h' is not visited yet, we walk around the target of 'h' and mark these
     // halfedges as visited. Thus, if we are here and the target is already marked as visited,
     // it means that the vertex is non manifold.
     if(halfedges_handled.insert(h).second)
     {
-      vertex_descriptor vd = target(h, tm);
-      if(!vertices_handled.insert(vd).second)
-      {
-        put(cmap, vd, true); // store the originals
-        non_manifold_cones.push_back(h);
-      }
-      else
-      {
-        set_halfedge(vd, h, tm);
-      }
+      bool is_non_manifold = false;
+
+      vertex_descriptor vd = target(h, pm);
+      if(!vertices_handled.insert(vd).second) // already seen this vertex, but not from this star
```

Comment:
We should replace both `vertices_handled` and `halfedges_handled` by dynamic properties

Reply:
Yes sure, I left it in other places where I thought the dynamic pmaps were not worth the memory because they would only be used on few elements, but here it should be.